### PR TITLE
User

### PR DESCRIPTION
**Pull request by Arique1104**
**Contributors:**Arique & Priya

**Details:** This PR will create a secure Users Database that ensures passwords are encrypted and emails are unique.

**Improvements:** (issues/bugs/refactoring?)

- SimpleCov Model (%): 7 / 28 LOC (25.0%)
- SimpleCov Feature (%): 5 / 27 LOC (18.52%)
- RuboCop (Complaints/Violations):

app/controllers/welcome_controller.rb:2:1: C: Layout/EmptyLinesAroundClassBody: Extra empty line detected at class body beginning.
app/controllers/welcome_controller.rb:3:3: C: Style/EmptyMethod: Put empty method definitions on a single line.
  def index ...
  ^^^^^^^^^
app/controllers/welcome_controller.rb:4:1: C: Layout/EmptyLinesAroundMethodBody: Extra empty line detected at method body beginning.
app/models/user.rb:3:3: C: Rails/Validation: Prefer the new style validations validates :column, presence: value over validates_presence_of.
  validates_presence_of :password, require: true
  ^^^^^^^^^^^^^^^^^^^^^
app/models/user.rb:4:22: C: Layout/TrailingWhitespace: Trailing whitespace detected.
  has_secure_password
                     ^
app/models/user.rb:5:1: C: Layout/EmptyLinesAroundClassBody: Extra empty line detected at class body end.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+RSpec.describe User, type: :model do
+  describe "validations" do
+    it {should validate_presence_of(:email)}
+    it {should validate_uniqueness_of(:email)}
+    it {should validate_presence_of(:password)}
+  end
+
+  describe "relationships" do
+  end
```
Comment (priyapower) [inline spec/models/user_spec.rb:11]: This is where we will begin as we do the self-referential joins for users and their "friends"

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -61,4 +61,12 @@
   config.filter_rails_from_backtrace!
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
+
+end
+
+Shoulda::Matchers.configure do |config|
```
Comment (priyapower) [inline spec/rails_helper.rb:67]: Glad we figured out the error was just a gem issue and not a break in our database!

**Review by priyapower — APPROVED** [review]
Looks great! I'm ready to merge!

File: spec/models/user_spec.rb (hunk 1 above)
Comment (Arique1104) [reply spec/models/user_spec.rb:11]: NOICE!

File: spec/rails_helper.rb (hunk 2 above)
Comment (Arique1104) [reply spec/rails_helper.rb:67]: Seriously glad we didn't break into our database!  Data no Databasing! (Swiper no Swiping!)

**Pull request merged**